### PR TITLE
FLUID-5944: Updating the README and ReleaseNotes for the Infusion 2.0 release

### PR DESCRIPTION
**Pull request by jobara**
https://issues.fluidproject.org/browse/FLUID-5944

### CODE REVIEW

File: ReleaseNotes.md [hunk 1]
```diff
@@ -1,92 +1,77 @@
...
+* Obsolete syntax for arguments, options, etc.
+* `"autoInit"` grade
+* Static and dynamic environments, replaced by Global Instantiator
+* The old model component hierarchy and "old ChangeApplier" implementation
+* `fluid.demands`
+* No more distinction between fast and dynamic invokers
+* Model Relay specific component grades have been removed, model relay now works with any model grade.
+
 
 ## What's New in 1.9 ##
```
Comment (michelled) [inline ReleaseNotes.md]: Do we want to keep this section? Isn't it good enough to have the what's new in 2.0?

Comment (jobara) [inline ReleaseNotes.md]: Typically we wouldn't want a section for a prior release in the release notes. However, we didn't actually release 1.9 so this includes those changes. We could move this info up to the What's new in 2.0 section though. What do you think?

File: README.md [hunk 2]
```diff
@@ -174,15 +173,35 @@ When this VM is first created Chrome and Firefox will be upgraded to the latest
...
+
+Please note that any browsers launched will need to remain in the foreground because some of the tests require window focus.
+
+### Run Tests Using Browsers Installed In a VM ###
+
+If installing Testem is not preferable, a [Fedora VM](https://github.com/idi-ops/packer-fedora) can be automatically created using tools provided by the [Prosperity4All Quality Infrastructure](https://github.com/GPII/qi-development-environments/). After meeting the [QI development VM requirements](https://github.com/GPII/qi-development-environments/#requirements) the ``vagrant up`` command can be used to launch a VM which will contain Testem and several browsers. Typing ``grunt tests`` will run the Infusion tests in the VM and the results will be displayed in your terminal.
```
Comment (amb26) [inline README.md]: preferable -> desirable

File: ReleaseNotes.md (hunk 1 above)
Comment (michelled) [inline ReleaseNotes.md]: I don't feel strongly either way. So I guess it's easier to just leave it as is - unless someone else has an opinion.

**Comment by jobara** [comment]
@amb26, @colinbdclark, @michelled ready for more review.

File: README.md [hunk 3]
```diff
@@ -174,15 +173,35 @@ When this VM is first created Chrome and Firefox will be upgraded to the latest
...
 
+## How Do I Run Tests? ##
+
+There are two options available for running tests. The first option involves using browsers installed on your computer and the second uses browsers available in a VM.
+
+### Run Tests Using Browsers Installed On Your Computer ###
+
+Using this option requires the installation of [Testem](https://github.com/testem/testem/#installation) and then running ``testem ci --file tests/testem.json`` in this directory. Any browsers that Testem finds on your platform will be launched sequentially with each browser running the full Infusion test suite. The results will be returned in your terminal in the [TAP](https://testanything.org/) format. You can use the ``testem launchers`` command to get a list of available browsers.
+
+Please note that any browsers launched will need to remain in the foreground because some of the tests require window focus.
```
Comment (colinbdclark) [inline README.md]: This statement could use clarification. If understand correctly, the user needs to actively monitor the tests as they run and focus each browser as it is opened. Is that correct?

File: README.md [hunk 4]
```diff
@@ -174,15 +173,35 @@ When this VM is first created Chrome and Firefox will be upgraded to the latest
...
+
+### Run Tests Using Browsers Installed In a VM ###
+
+If installing Testem is not desirable, a [Fedora VM](https://github.com/idi-ops/packer-fedora) can be automatically created using tools provided by the [Prosperity4All Quality Infrastructure](https://github.com/GPII/qi-development-environments/). After meeting the [QI development VM requirements](https://github.com/GPII/qi-development-environments/#requirements) the ``vagrant up`` command can be used to launch a VM which will contain Testem and several browsers. Typing ``grunt tests`` will run the Infusion tests in the VM and the results will be displayed in your terminal.
```
Comment (colinbdclark) [inline README.md]: Are there other reasons why testing in a VM is preferable, aside from not having to install testem?

File: ReleaseNotes.md [hunk 5]
```diff
@@ -1,224 +1,348 @@
...
-
-## What's New in 1.9 ##
-
-**Note** 1.9 was never (so far) an official release of Infusion, but was a "last ditch" branching point to record the state of the framework should
-we ever in future require to make a release or backport fixes to a version of Infusion API compatible with the latest 1.x line, most particularly 1.5.x.
-It is stored in github at [Infusion 1.9.x](https://github.com/fluid-project/infusion/tree/1.9.x)
-
-This list of JIRAs is nonexhaustive since work has accumulated in trunk for several years since the 1.5 Infusion release. These are some of the more significant recent fixes,
-currently not including work on the preferences framework:
+**Note:** Infusion 1.9 was never (so far) an official release of Infusion, but was a "last ditch" branching point to record the state of the framework should
```
Comment (colinbdclark) [inline ReleaseNotes.md]: I don't really know what this "'last ditch' branching point" statement means, and I don't think many users will be able to interpret it. Can we say this is in simpler terms?

"Infusion 1.9 was not officially released, but is available as an official branch so that we can backport fixes to a version of the Infusion API compatible with the latest 1.x line."

Or something along these lines?

File: ReleaseNotes.md [hunk 6]
```diff
@@ -1,224 +1,348 @@
...
+we ever require to make a release or backport fixes to a version of the Infusion API compatible with the latest 1.x line.
+It is stored in a branch on GitHub at [Infusion 1.9.x](https://github.com/fluid-project/infusion/tree/1.9.x). For a complete list of Fixes and Improvements see the [Version 1.9](https://issues.fluidproject.org/projects/FLUID/versions/10520) summary in the [JIRA](https://issues.fluidproject.org) issue tracker.
 
 ### New Features
 
-* [FLUID-5513: Implement "micropromises" to ease manipulation of asynchronous code sequences](https://issues.fluidproject.org/browse/FLUID-5513)
+* Constraint-based priorities, supported by `listeners`, `modelListeners`, `modelRelay`, `distributeOptions`, `contextAwareness`, and `components`
```
Comment (colinbdclark) [inline ReleaseNotes.md]: What does this feature provide, very briefly?

File: ReleaseNotes.md [hunk 7]
```diff
@@ -1,224 +1,348 @@
...
+* Context Awareness - and things it relies on:
+  * Global Instantiator
+    * Every Infusion component, regardless of how it is instantiated, ends up in a single-rooted tree of components
+    * This enables use of modern IoC features such as model relay and declarative event binding
+    * Enables use of the root distributeOptions context "/"
+    * Enables the removal of "demands blocks"
+    * Useful debugging tip: Watch `fluid.globalInstantiator` in your JS debugging tools to see the structure of your application and its tree.
+* `fluid.notImplemented` function for implementing abstract grades
+* [Lazy loading UI Options](http://docs.fluidproject.org/infusion/development/UserInterfaceOptionsAPI.html#lazyload) and instructions for how to use the Prefernces Framework with a [minimal footprint](http://docs.fluidproject.org/infusion/development/tutorial-prefsFrameworkMinimalFootprint/MinimalFootprint.html).
```
Comment (colinbdclark) [inline ReleaseNotes.md]: "Lazy loading for UI Options"

Typo Prefernces -> "Preferences"

What does "minimal footprint" mean? Can we instead describe what this actually means for users? Is it "zero initial load time" or something like that?

File: ReleaseNotes.md [hunk 8]
```diff
@@ -1,224 +1,348 @@
...
+  * This should assist in improving performance when using the Preferences Framework, particularly for resource intensive sites and applications
+* Much faster invokers and boiled listeners (c. 60x faster)
+* NPM - "dists" directory containing pre-built versions of Infusion
```
Comment (colinbdclark) [inline ReleaseNotes.md]: Change this to something like:
  
    * Support for using Infusion with npm for both Node.js and web-based projects. This includes a variety of prebuilt versions of Infusion included in the module's <code>dist</code> directory.

File: ReleaseNotes.md [hunk 9]
```diff
@@ -1,224 +1,348 @@
...
+* Source Maps are generated for the concatenated JavaScript files
+* View oriented IoC debugging tools
+  * FluidViewDebugging.js - include this at the head of your HTML file to get access to the "IoC inspector"
```
Comment (colinbdclark) [inline ReleaseNotes.md]: What is this (even in just a few words)? Is there a link to learn anything about it? Why is it in quotes?

File: ReleaseNotes.md [hunk 10]
```diff
@@ -1,224 +1,348 @@
...
 
-### Bug Fixes and Improvements
+### Removals
```
Comment (colinbdclark) [inline ReleaseNotes.md]: Should this be "Removal of Deprecated Features"?

File: ReleaseNotes.md [hunk 11]
```diff
@@ -1,224 +1,348 @@
...
-find them in the _**demos**_ folder in the release bundle or on our [build site](http://build.fluidproject.org/).
+Infusion ships with demos of all of the components in action. You can find them in the _**demos**_ folder in the release bundle or on our [build site](http://build.fluidproject.org/).
 
-When run from a local file system, several of these demos require you to enable local file AJAX
-in Firefox and Chrome:
-
-* https://wiki.fluidproject.org/display/fluid/Browser+settings+to+support+local+Ajax+calls
-* http://kb.mozillazine.org/Security.fileuri.strict_origin_policy
-* http://ejohn.org/blog/tightened-local-file-security/
+When running from your local machine, it is recommended to use a web server for accesing the demos. Several of the demos make use of AJAX calls; which typically are not allowed by
```
Comment (colinbdclark) [inline ReleaseNotes.md]: "When running the demos on your local machine, a web server is recommended."

File: ReleaseNotes.md [hunk 12]
```diff
@@ -1,224 +1,348 @@
...
-
-### ECL 2.0 ###
-* Sample markup and stylesheets from [Sakai v2.5](http://sakaiproject.org)
-
-### Apache 2.0 ###
-* [Open Sans Light font](http://www.google.com/fonts/specimen/Open+Sans)
 
 ### Public Domain ###
+
 * Douglas Crockford's [JSON.js (from 2007-11-06)](http://www.json.org/)
```
Comment (colinbdclark) [inline ReleaseNotes.md]: As far as I can tell, we no long distribute Crockford's old JSON parser with Infusion, right?

File: ReleaseNotes.md [hunk 13]
```diff
@@ -1,224 +1,348 @@
...
 
-We are in the process of migrating our documentation to a new space. The markdown files for the documentation can be found in [github](https://github.com/fluid-project/infusion-docs).
+* [fastXmlPull is based on XML for Script's Fast Pull Parser v3.1](http://wiki.fluidproject.org/display/fluid/Licensing+for+fastXmlPull.js)
 
-The new space is dedicated to only Infusion documentation, and provides improved navigation.
+## Documentation ##
 
-Some of our documentation remains in the wiki space: Links to these pages are indicated with _**wiki**_. From any of these pages, you can return to the main documentation space using your
-browser's Back button.
+Documentation and tutorials can found on the [Infusion Documentation](http://docs.fluidproject.org/infusion/development/) site.
```
Comment (colinbdclark) [inline ReleaseNotes.md]: If we haven't already, can we make sure that all of the old documentation pages in our wiki have had their content removed and prominent redirects or notices placed on them to the docs site?

Comment (jobara) [inline ReleaseNotes.md]: Discussion in the [fluid-work IRC channel](https://botbot.me/freenode/fluid-work/2016-12-02/?msg=77365799&page=1)

File: README.md [hunk 14]
```diff
@@ -32,7 +33,7 @@ The best way to join the Fluid Community is to jump into any of our community ac
 
 ## Where is Infusion Used? ##
 
-The Infusion library is the cornerstone of a number of projects. You can see some of them featured on our [Projects page](http://fluidproject.org/projects.html).
+The Infusion library is the cornerstone of a number of projects. You can see some of them featured on our [Projects page](http://fluidproject.org/projects.html). Additional integrations of Infusion can be found on the [Infusion Integrations](https://wiki.fluidproject.org/display/fluid/Infusion+Integrations) wiki page.
```
Comment (jobara) [inline README.md]: sounds good, i'll use your suggestion.

File: README.md (hunk 3 above)
Comment (jobara) [inline README.md]: Yes, the browser must have focus.

Comment (colinbdclark) [inline README.md]: Okay, then I think we should describe what the user actually has to do to ensure the tests will pass—tell them explicitly that they will need to focus each browser.

**Comment by jobara** [comment]
@colinbdclark ready for more review

**Comment by jobara** [comment]
merged at 73f7f90d0a0e1a52b0aac31b0ee941e8835df8cd
